tests/bluetooth/tester/src/btp.h

```diff
@@ -0,0 +1,102 @@
+/* bttester.h - Bluetooth tester headers */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/util.h>
+#include <zephyr/bluetooth/addr.h>
+
+#define BTP_MTU 1024
+#define BTP_DATA_MAX_SIZE (BTP_MTU - sizeof(struct btp_hdr))
+
+#define BTP_INDEX_NONE		0xff
+
+#define BTP_SERVICE_ID_CORE	0
+#define BTP_SERVICE_ID_GAP	1
+#define BTP_SERVICE_ID_GATT	2
+#define BTP_SERVICE_ID_L2CAP	3
+#define BTP_SERVICE_ID_MESH	4
+
+#define BTP_STATUS_SUCCESS	0x00
+#define BTP_STATUS_FAILED	0x01
+#define BTP_STATUS_UNKNOWN_CMD	0x02
+#define BTP_STATUS_NOT_READY	0x03
+
+struct btp_hdr {
+	uint8_t  service;
+	uint8_t  opcode;
+	uint8_t  index;
+	uint16_t len;
+	uint8_t  data[];
+} __packed;
+
+#define BTP_STATUS			0x00
+struct btp_status {
+	uint8_t code;
+} __packed;
+
+/* Core Service */
+#define CORE_READ_SUPPORTED_COMMANDS	0x01
+struct core_read_supported_commands_rp {
+	uint8_t data[0];
+} __packed;
+
+#define CORE_READ_SUPPORTED_SERVICES	0x02
+struct core_read_supported_services_rp {
+	uint8_t data[0];
+} __packed;
+
+#define CORE_REGISTER_SERVICE		0x03
+struct core_register_service_cmd {
+	uint8_t id;
+} __packed;
+
+#define CORE_UNREGISTER_SERVICE		0x04
+struct core_unregister_service_cmd {
+	uint8_t id;
+} __packed;
+
+/* events */
+#define CORE_EV_IUT_READY		0x80
+
+static inline void tester_set_bit(uint8_t *addr, unsigned int bit)
+{
+	uint8_t *p = addr + (bit / 8U);
+
+	*p |= BIT(bit % 8);
+}
+
+static inline uint8_t tester_test_bit(const uint8_t *addr, unsigned int bit)
+{
+	const uint8_t *p = addr + (bit / 8U);
+
+	return *p & BIT(bit % 8);
+}
+
+void tester_init(void);
+void tester_rsp(uint8_t service, uint8_t opcode, uint8_t index, uint8_t status);
+void tester_send(uint8_t service, uint8_t opcode, uint8_t index, uint8_t *data,
+		 size_t len);
+
+uint8_t tester_init_gatt(void);
+uint8_t tester_unregister_gatt(void);
+void tester_handle_gatt(uint8_t opcode, uint8_t index, uint8_t *data,
+			uint16_t len);
+
+uint8_t tester_init_l2cap(void);
+uint8_t tester_unregister_l2cap(void);
+void tester_handle_l2cap(uint8_t opcode, uint8_t index, uint8_t *data,
+			 uint16_t len);
+
+uint8_t tester_init_mesh(void);
+uint8_t tester_unregister_mesh(void);
+void tester_handle_mesh(uint8_t opcode, uint8_t index, uint8_t *data, uint16_t len);
+
+uint8_t tester_init_gap(void);
+uint8_t tester_unregister_gap(void);
+void tester_handle_gap(uint8_t opcode, uint8_t index, uint8_t *data,
+		       uint16_t len);
```

Comment:
I'd leave those in `bttester.h` as those are loosely related to BTP

Reply:
Done